### DISCUSSION TITLE
CLOUDFRONT_CDN Question

### DISCUSSION THREAD

**Discussion by Simoon-F** [Budibase/budibase#9755] [Help] [answered]
Do I have to set CLOUDFRONT_PRIVATE_KEY_64 andCLOUDFRONT_PUBLIC_KEY_ID after setting the CLOUDFRONT_CDN? I am not using AWS。

![image](https://user-images.githubusercontent.com/48503662/220226104-e83d58c9-a696-4d3d-a9c0-c9ee179e48b5.png)

There is also a very strange problem. My local Docker setting CDN Link does not have this prompt, but the online k8s has this prompt, This makes me very confused...

**Comment by Simoon-F** [comment]
The seven cattle cloud CDN I use, I want to use it..

**Comment by Rory-Powell** [comment, marked as answer]
Hi @Simoon-F 

The `CLOUDFRONT_CDN` environment variable is specifically tailored to using Cloudfront CDN on AWS. We use this in combination with the private and public keys you mentioned, with a signing library: https://www.npmjs.com/package/aws-cloudfront-sign. 

Using a CDN other than Cloudfront is not supported right now. 

The built-in configuration will serve files from Mino via the nginx proxy, or by S3 directly if configured instead of Minio. If this doesn't suit your needs you can raise a feature request to support a custom CDN.
